README.md

```diff
@@ -1,2 +1,39 @@
+- Benchmark parameters
+  - This may be an array of multiple parameters, when the user wants the benchmark run mulitple ways, for example:
+    <pre>[ "--rw=read --bs=4k", "--rw=write --bs=4k" ]</pre>
+  - The [multiplex](https://github.com/perftool-incubator/multiplex) project can be used to generate this array (it can convert things like "--rw=read,write --bs=4k" into the array above), and it will do parameter validation for you as well.
+- Endpoint extension
+  - This determines how the benchmark gets executed on different endpoints.  The default extension, local, simply runs the benchmark command on the local host.  Rickshaw supports using multiple extensions for different endpoints.  For example, if you want to run uperf benchmark, you need both a client and server uperf.  If you want to run the uperf server on Kuberbetes, but you want to run the uperf client on a baremetal host, you can use the 'k8s' extension for the server and the 'ssh' extension for the client
+  - <pre>--endpoint:server[1]:k8s:$master-hostname --endpoint:client[1]:ssh:$client-hostname</pre>
+  - Other examples
+    - 8 servers running in 8 containers in k8s and 8 clients running on the same barmetal host:  
+      - <pre>--endpoint:server[1-8]:k8s:$master-hostname --endpoint:client[1-8]:ssh:$client-hostname</pre>
+    - 8 servers running in 8 containers in k8s cluster A and 8 clients running in 8 containers in k8s cluster B:  
+      - <pre>--endpoint:server[1-8]:k8s:$cluster-a-master-hostname --endpoint:client[1-8]:k8s:$cluster-b-master-hostname</pre>
+  - If you don't provide any endpoint options, rickshaw will assume you want to run on the local system, and only 1 client will be used, and if the benchmark requires a server, only 1 server will be used, also on the local host.  
+- User and test information
+  - The following is optional, but is highly recommended to use so that your run can be easily searchable later.  Rickshaw looks for the following
+    - RS_NAME Environment variable or --name option with user's full name "First Last"
+    - RS_EMAIL Environment variable or --email option with user's email address "my-email@email.domain"
+    - RS_TAGS Environment variable or --tags containing a comma-separated list of words that are relevant to the run
+    - RS_DESC Environment variable or --desc containing a free form description of the purpose, conditions, or any other relevant information about this test.
+    
+  
+### Output
+  
+Rickshaw will provide the following:
+- Human readable log of the rickshaw execution, including any warning and errors from other projects, tools, or benchmarks it used.
+- A JSON file or Elastic documents describing the benchmark run that was handled by rickshaw
```

Comment:
I had not really considered both a JSON file AND the Elastic CDM documents as being output but I suppose that makes sense.  Would the JSON file be an intermediary format used to create the CDM documents?

Reply:
I was just thinking that users trying to parse multiple CDM docs might be difficult, and so having a JSON doc which at least captures the run data (run/iteration/sample/period docs, but not the metric docs) would be nice.  I think we can make it so we use this JSON doc to generate the ES docs.  We could even go straight from this JSON doc to importing into ES (many docs) and skip the local-generation of many CDM docs.

It would also be a lot easier IMO to change something like the run.description in the JSON by hand if you wanted to.

There's a possibility we could do the same for metrics, but I would table that until we are certain it works well for the run/iteration/sample/period docs.